cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceTaskContextProvider.java

```diff
@@ -207,10 +229,15 @@ public BasicMapReduceTaskContext load(ContextCacheKey key) throws Exception {
     private final Configuration configuration;
 
     private ContextCacheKey(TaskAttemptContext context) {
-      this.taskAttemptID = context.getTaskAttemptID();
-      this.configuration = context.getConfiguration();
+      this(context.getTaskAttemptID(), context.getConfiguration());
+    }
+
+    private ContextCacheKey(@Nullable TaskAttemptID taskAttemptID, Configuration configuration) {
```

Comment:
Right - for partitioner or comparator, there will be no metrics collections.

In Standalone then, the Partitioners and Comparators would share a MapReduceTaskContext. I don't see how that would be an issue.